sdk/iot/Azure.Iot.Hub.Service/src/Authentication/SharedAccessSignatureBuilder.cs

```diff
@@ -0,0 +1,84 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Net;
+using System.Security.Cryptography;
+using System.Text;
+
+namespace Azure.Iot.Hub.Service.Authentication
+{
+    internal sealed class SharedAccessSignatureBuilder
+    {
+        public SharedAccessSignatureBuilder()
+        {
+            TimeToLive = TimeSpan.FromMinutes(5);
+        }
+
+        public string KeyName { get; set; }
+
+        public string Key { get; set; }
+
+        public string Target { get; set; }
+
+        public TimeSpan TimeToLive { get; set; }
+
+        public string ToSignature()
+        {
+            return BuildSignature(KeyName, Key, Target, TimeToLive);
+        }
+
+        private static string BuildSignature(string keyName, string key, string target, TimeSpan timeToLive)
```

Comment:
I know unit tests are coming in a separate PR, just wanted to emphasize that we need extensive unit tests for this fella